app/code/Magento/InventoryCatalog/Test/Api/StockSourceLink/PreventAssignSourcesToDefaultStockTest.php

```diff
@@ -77,24 +91,63 @@ public function dataProviderWrongParameters(): array
     {
         $defaultSourceProvider = Bootstrap::getObjectManager()->get(DefaultSourceProviderInterface::class);
         $defaultStockProvider = Bootstrap::getObjectManager()->get(DefaultStockProviderInterface::class);
-
         return [
             'multiple_sources_assigned_to_default_stock' => [
                 [$defaultSourceProvider->getCode(), 'eu-2'],
                 $defaultStockProvider->getId(),
                 [
-                    'rest_message' => 'You can only assign Default Source to Default Stock',
-                    'soap_message' => 'You can only assign Default Source to Default Stock',
+                    'message' => 'Validation Failed',
+                    'errors' => [
+                        [
+                            'message' => 'Can not save link related to Default Source or Default Stock',
+                            'parameters' => [],
+                        ],
+                        [
+                            'message' => 'Can not save link related to Default Source or Default Stock',
+                            'parameters' => [],
+                        ],
+                    ],
                 ],
             ],
             'not_default_source_assigned_to_default_stock' => [
                 ['eu-1'],
                 $defaultStockProvider->getId(),
                 [
-                    'rest_message' => 'You can only assign Default Source to Default Stock',
-                    'soap_message' => 'You can only assign Default Source to Default Stock',
+                    'message' => 'Validation Failed',
+                    'errors' => [
+                        [
+                            'message' => 'Can not save link related to Default Source or Default Stock',
+                            'parameters' => [],
+                        ],
+                    ],
                 ],
             ],
         ];
     }
+
+    /**
+     * @inheritdoc
+     */
+    protected function _checkWrappedErrors($expectedWrappedErrors, $errorDetails)
```

Comment:
why to add protected method with name starting from underscore to the newly created code?

Reply:
@maghamed, as original _checkWrappedErrors():
https://github.com/magento-engcom/msi/blob/f2fdae5246c1b859e4ce7532458dd1243eff0991/dev/tests/api-functional/framework/Magento/TestFramework/TestCase/WebapiAbstract.php#L639
 can not handle ValidationException without detail node thrown by https://github.com/magento-engcom/msi/blob/f2fdae5246c1b859e4ce7532458dd1243eff0991/app/code/Magento/Inventory/Model/StockSourceLink/Command/StockSourceLinksSave.php#L68 I had to rewrite this paricular method with my own.